Files/Interacts/Interaction.cs

```diff
@@ -429,98 +430,144 @@ public async void OpenFileLocation_Click(object sender, RoutedEventArgs e)
             }
         }
 
-        private async void OpenSelectedItems(bool displayApplicationPicker)
+        /// <summary>
+        /// Navigates to a directory or opens file
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="itemType"></param>
+        /// <param name="openSilent">Determines whether history of opened item is saved (... to Recent Items/Windows Timeline/opening in background)</param>
+        /// <param name="openViaApplicationPicker">Determines whether open file using application picker</param>
+        public async Task<bool> OpenPath(string path, FilesystemItemType? itemType = null, bool openSilent = false, bool openViaApplicationPicker = false)
+        // TODO: This function reliability has not been extensively tested
         {
-            if (AssociatedInstance.FilesystemViewModel.WorkingDirectory.StartsWith(AppSettings.RecycleBinPath))
+            string previousDir = AssociatedInstance.FilesystemViewModel.WorkingDirectory;
+            bool isHiddenItem = NativeFileOperationsHelper.HasFileAttribute(path, System.IO.FileAttributes.Hidden);
+            bool isShortcutItem = path.EndsWith(".lnk") || path.EndsWith(".url"); // Determine
+            bool fileExists = await StorageItemHelpers.Exists(path, AssociatedInstance);
+            FilesystemResult opened = (FilesystemResult)false;
+
+            // Shortcut item variables
+            string sTargetPath = null;
+            string sArguments = null;
+            string sWorkingDirectory = null;
+            bool sRunAsAdmin = false;
+            bool sIsFolder = false;
+
+            if (!fileExists && !isShortcutItem && !isHiddenItem)
```

Comment:
[Code style](https://github.com/files-community/Files/wiki/Code-style#11)
About variable names, do not use one letter prefix, name it targetPath or shortcutTargetPath.

Reply:
Ok, done 😊

Reply:
Can you add curly brackets, please? And in other places too.